tools/kagami/src/genesis/sign.rs

```diff
@@ -0,0 +1,73 @@
+use std::{
+    fs,
+    fs::File,
+    io::{BufWriter, Write},
+    path::PathBuf,
+};
+
+use clap::{ArgGroup, Parser};
+use color_eyre::eyre;
+use iroha_crypto::{KeyPair, PrivateKey};
+use iroha_data_model::prelude::*;
+use iroha_genesis::RawGenesisTransaction;
+use parity_scale_codec::Encode;
+
+use crate::{Outcome, RunArgs};
+
+/// Sign the genesis block
+#[derive(Clone, Debug, Parser)]
+#[command(group = ArgGroup::new("private_key_group").required(true))]
+#[command(group = ArgGroup::new("public_key_group").required(true))]
+pub struct Args {
+    /// Path to genesis json file
+    genesis_file: PathBuf,
+    /// Genesis private key
+    #[clap(long, group = "private_key_group", value_name("MULTIHASH"))]
+    private_key: Option<PrivateKey>,
+    /// Genesis public key
+    #[clap(long, group = "public_key_group", value_name("MULTIHASH"))]
+    public_key: Option<PublicKey>,
+    /// Path to json-serialized `KeyPair`
+    #[clap(
+        long,
+        group = "private_key_group",
+        group = "public_key_group",
+        value_name("PATH")
+    )]
+    keypair_file: Option<PathBuf>,
+    /// Path to signed genesis output file in SCALE format (stdout by default)
+    #[clap(short, long, value_name("PATH"))]
+    out_file: Option<PathBuf>,
```

Comment:
Not entirely related, but please also rename `--outfile` to `--out-file` for `iroha_swarm`.

Reply:
#4675